Source/drlg_l3.cpp

```diff
@@ -424,8 +424,8 @@ static void DRLG_L3River()
 			ry = 0;
 			i = 0;
 			while ((dungeon[rx][ry] < 25 || dungeon[rx][ry] > 28) && i < 100) {
-				rx = random_(0, DMAXX);
-				ry = random_(0, DMAXY);
+				rx = random_(0, DMAXX); // BUGFIX OOB `DMAXX - 1`
+				ry = random_(0, DMAXY); // BUGFIX OOB `DMAXY - 1`

```

Comment:
why? random_(0,DMAXX) returns 0 - (DMAXX-1), see the definition of int random_(BYTE idx, int v)